BeVolt/Apps/Src/CLI.c

```diff
@@ -1,774 +1,780 @@
+					case DANGER: 
+						printf("DANGER\n\r");
+						break;
+					case OVERVOLTAGE:
+						printf("OVERVOLTAGE\n\r");
+						break;
+					case UNDERVOLTAGE: 
+						printf("UNDERVOLTAGE\n\r");
+						break;
+					default:
+						break;
+				}		
+				break;
+		default:
+			printf("Invalid voltage command\n\r");
+			break;
+	}
+}
+
+/** CLI_Current
+ * Checks and displays the desired
+ * current parameter(s)
+ * @param hashTokens is the array of hashed tokens
+ */
+void CLI_Current(int* hashTokens) {
+	if(hashTokens[1] == 0) {
+		printf("High: %.3fA\n\r", Current_GetHighPrecReading()/MILLI_UNIT_CONVERSION);	// Prints 4 digits, number, and A
+		printf("Low: %.3fA\n\r", Current_GetLowPrecReading()/MILLI_UNIT_CONVERSION);
+		return;
+	}
+	switch (hashTokens[1]) {
+		case CLI_HIGH_HASH: // High precision reading
+			printf("High: %.3fA     \n\r", Current_GetHighPrecReading()/MILLI_UNIT_CONVERSION);
+			break;
+		case CLI_LOW_HASH: // Low precision reading
+			printf("Low: %.3fA     \n\r", Current_GetLowPrecReading()/MILLI_UNIT_CONVERSION);
+			break;
+		case CLI_SAFE_HASH: 
+		case CLI_SAFETY_HASH:
+			if (Current_CheckStatus(false) == SAFE) {
+				printf("Safety Status: SAFE\n\r");
+			}
+			else {
+				printf("Safety Status: DANGER\n\r");
+			}
+			break;
+		case CLI_CHARGE_HASH: // Whether battery is charging
+		case CLI_CHARGING_STATE:
+			if (Current_IsCharging()) {
+				printf("Charging State: CHARGING\n\r");
+			}
+			else {
+				printf("Charging State: DISCHARGING\n\r");
+			}
+			break;
+		default:
+			printf("Invalid Current Command\n\r");
+			break;
+		}
+}
+
+/** CLI_Temperature
+ * Checks and displays the desired
+ * temperature parameter(s)
+ * @param hashTokens is the array of hashed tokens
+ */
+void CLI_Temperature(int* hashTokens) {
+	if(hashTokens[1] == 0) {
+		for(int i = 0; i < NUM_BATTERY_MODULES; i++) {
+			printf("Module number %d: %.3f C\n\r", i+1, Temperature_GetModuleTemperature(i)/MILLI_UNIT_CONVERSION);
+		}
+		return;
+	}
+	switch(hashTokens[1]) {			
+		// All temperature sensors
+		case CLI_ALL_HASH:
+			for(int i = 0; i < NUM_MINIONS; i++) {		// last minion only has 14 sensors
+				for(int j = 0; j < MAX_TEMP_SENSORS_PER_MINION_BOARD; j++) {
+					if(i == NUM_MINIONS-1 && j > NUM_TEMP_SENSORS_LAST_MINION-1) {	// -1 for 0-index
+						break;
+					}
+					printf("Sensor number %d: %.3f C\n\r",(i*MAX_TEMP_SENSORS_PER_MINION_BOARD)+j+1, Temperature_GetSingleTempSensor(i, j)/MILLI_UNIT_CONVERSION);
+				}
+			}
+			break;
+		// Temperature of specific module
+		case CLI_MODULE_HASH:
+			if (hashTokens[2] == 0 || hashTokens[2]-1 > NUM_BATTERY_MODULES || hashTokens[2]-1 < 0){
+				printf("Invalid module number\n\r");
+			}
+			else {
+				if(hashTokens[3] == 0) {//temperature of module
+					printf("Module number %d: %.3f C\n\r", hashTokens[2], Temperature_GetModuleTemperature(hashTokens[2]-1)/MILLI_UNIT_CONVERSION);
+				} else if(hashTokens[3]-1 == 0 || hashTokens[3]-1 == 1) {//temperature of specific sensor in module
```

Comment:
Why are we subtracting one and then comparing to 0 and 1? Why not `hashTokens[3] == 1` and `hashTokens[3] == 2`? In fact, since I'm not sure what 1 and 2 would mean for a hash token, why not use some sort of enum (if these are status codes)?

I realize this is probably out of scope for the contactor simulator pull, but we should address this at some point.

@JimothyGreene what's going on here?

Reply:
That's just a sensor number (0 or 1). I was just trying to keep the 0-indexing on the code side and the 1-indexing on the user input side. I can see how this is confusing, so it can be changed.

Reply:
Yeah I think that we could either change the user input side to 0-indexed or change it so that we have definitions/constants like `SENSOR_ID_1 = 0`. Both would make it easier to read.